angularjs/angular.d.ts

```diff
@@ -1413,13 +1402,15 @@ declare module angular {
         (data: any, headersGetter: IHttpHeadersGetter, status: number): any;
     }
 
+    type HttpHeaderType = {[requestType: string]:string|((config:IRequestConfig) => string)};
+
     interface IHttpRequestConfigHeaders {
-        [requestType: string]: string|(() => string);
-        common?: string|(() => string);
-        get?: string|(() => string);
-        post?: string|(() => string);
-        put?: string|(() => string);
-        patch?: string|(() => string);
+        [requestType: string]: any;
+        common?: any;
+        get?: any;
+        post?: any;
+        put?: any;
+        patch?: any;
```

Comment:
Not sure if I like this - the way they put it we would rather have two distinct types for the two different usages, and see which one is appropriate here. I will need to take a closer look at the other issue and the source, though, before I can give a final feedback.